paddle/fluid/operators/mkldnn/matmul_mkldnn_op.cc

```diff
@@ -329,10 +329,16 @@ class MatMulMKLDNNHandler
   }
 
   bool IsOutputFused(const ExecutionContext& ctx) const {
-    auto& fused_reshape_Out = ctx.Attr<std::vector<int>>("fused_reshape_Out");
-    auto& fused_transpose_Out =
-        ctx.Attr<std::vector<int>>("fused_transpose_Out");
-    return !fused_reshape_Out.empty() && !fused_transpose_Out.empty();
+    if (ctx.HasAttr("fused_transpose_Out") &&
+        ctx.HasAttr("fused_reshape_Out")) {
+      auto axis = ctx.Attr<std::vector<int>>("fused_transpose_Out");
+      auto shape = ctx.Attr<std::vector<int>>("fused_reshape_Out");
+
+      if (!axis.empty() && !shape.empty()) {
```

Comment:
This pattern is pretty common here, isn't it?
Please create a helper function which can be reused.

Reply:
I think, we can remove this whole block that was checking if attribute is empty. That's because after removing attributes from base op, we can just check if they exists. Enforces in fuse pass don't let setting empty values.